### PR TITLE
fix(npm): eagerly reload package information when version from lockfile not found locally

### LINKED ISSUE #18624: Eagerly reload package information from npm registry when package version in lockfile not found in cache
Right now it will error with:

```
Could not find '<package-name>' specified in the lockfile. Maybe try again with --reload
```

We should just have it reload in this case.

### PR DESCRIPTION
**Pull request by dsherret**
Closes #18624

### CODE REVIEW

**Review by bartlomieju — APPROVED** [review]
LGTM 🚀

File: cli/args/lockfile.rs [hunk 1]
```diff
@@ -114,25 +115,45 @@ pub async fn snapshot_from_lockfile(
...
-    packages[i].dist = version_info?.dist;
+  while let Some(result) = version_infos.next().await {
+    packages[i].dist = match result {
+      Ok(version_info) => version_info.dist,
+      Err(err) => {
+        if api.mark_force_reload() {
+          // reset and try again
+          version_infos = get_version_infos();
+          i = 0;
+          continue;
```
Comment (bartlomieju) [inline cli/args/lockfile.rs:144]: Actually looking at it once again - won't this cause infinite loop?

**Review by bartlomieju — CHANGES_REQUESTED** [review]
See comment above

File: cli/args/lockfile.rs (hunk 1 above)
Comment (dsherret) [reply cli/args/lockfile.rs:144]: Nope. `mark_force_reload` will only return `true` once.

Comment (dsherret) [reply cli/args/lockfile.rs:144]: Maybe I'm missing what you're seeing though. Could you elaborate?

**Review by bartlomieju — APPROVED** [review]

**Pull request merged**